mesheryctl/pkg/utils/helpers_test.go

```diff
@@ -0,0 +1,504 @@
+package utils
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+func getFixturesDirectory() string {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		log.Fatal("TestBackupConfigFile: Cannot get current working directory")
+	}
+	currDir := filepath.Dir(filename)
+	// get the fixtures file directory
+	fixturesDir := filepath.Join(currDir, "fixtures")
+	return fixturesDir
+}
+
+var fixturesDir = getFixturesDirectory()
+
+func TestBackupConfigFile(t *testing.T) {
+	name := "config.yaml"
+	configFilePath := filepath.Join(fixturesDir, name)
+
+	backupFileName := "config.bak.yaml"
+	backupConfigFilePath := filepath.Join(fixturesDir, backupFileName)
+
+	// creates a config file
+	NewGoldenFile(t, name, fixturesDir).Write("mesheryctl")
+
+	BackupConfigFile(configFilePath)
+
+	// check if backup file is present or not
+	_, err := os.Stat(backupConfigFilePath)
+	if err != nil {
+		t.Errorf("BackupConfigFile error = %v", err)
+	}
+}
+
+func TestStringWithCharset(t *testing.T) {
+	// checking the length, since this function returns random strings everytime
+	strLength := 10
+
+	gotString := StringWithCharset(strLength)
+	actualStringLength := len(gotString)
+
+	if actualStringLength != strLength {
+		t.Errorf("StringWithCharset got = %v want = %v", actualStringLength, strLength)
+	}
+}
+
+func TestPrereq(t *testing.T) {
+	_, _, err := prereq()
+	if err != nil {
+		t.Errorf("prereq error = %v", err)
+	}
+}
+
+func TestSetFileLocation(t *testing.T) {
+	err := SetFileLocation()
+	if err != nil {
+		t.Errorf("SetFileLocation error = %v", err)
+	}
+}
+
+// func TestNavigateToBrowser(t *testing.T) {
+// 	// opens up a browser window whenever this test runs
+// 	err := NavigateToBrowser("https://www.layer5.io")
+// 	if err != nil {
+// 		t.Errorf("NavigateToBrowser error: %v", err)
+// 	}
+// }
```

Comment:
Had to comment this test because it opens the browser, which I think will fail in automated test environment. Let me know how can I fix this.

Reply:
@piyushsingariya What should I do about this?

Reply:
We can run it tbh, just to add it to the code-coverage.

Reply:
![image](https://user-images.githubusercontent.com/54525741/133831752-c3fd97c6-bde5-406a-86a9-607b92846463.png)

That failed :(

Reply:
@piyushsingariya What should I do there?

Reply:
Could you check if the function has been already covered under some other test in codecov's UI? If Yes we can skip this dedicated test, if not then we gotta improve the github workflow a bit to add this to coverage.

Reply:
![image](https://user-images.githubusercontent.com/54525741/134013924-8c8737b3-84bb-4a28-b023-9d074691485f.png)
Doesn't look like they are covered. So, what changes should we make in actions for this?

Reply:
:+1: looking into it

Reply:
So I'm guessing we have to install a browser for this like this: https://github.com/meshery/meshery/blob/33e27c71ce87400389bbe8369e6ef308ebe83949/.github/workflows/go-testing-ci.yml#L51-L52

Reply:
@piyushsingariya Would it be fine to add a browser such as `firefox`?

Not sure why `lynx` is not working(since it is a text-based browser)

Reply:
as this asked for a method to open the browser, we gotta improve the CI workflow here, look at the integration tests workflow, it should resolve this issue.

Reply:
Yeah, so I was looking at the workflow output logs, and It does look for `lynx` but for some reason it cannot find it.
![image](https://user-images.githubusercontent.com/54525741/134643115-7c714f1f-d51b-4a04-87e1-47a508ec6c68.png)

Also, the `Install lynx for xdg-open support` step isn't running for some reason. So, I have to troubleshoot that.

Reply:
Oh, it turns our unit-tests doesn't have the step to install lynx, so I have to add that.

@piyushsingariya Should I make a new PR to add `lynx` installation in unit-tests CI?

Reply:
in the same PR.